Docs for snap tools (#10458)
opencv/cvat: site/content/en/docs/annotation/manual-annotation/shapes/annotation-with-polygons/snap-tools.md

---
title: "Snap tools"
linkTitle: "Snap tools"
weight: 2
aliases:
- /docs/manual/advanced/annotation-with-polygons/snap-tools/

> - /docs/annotation/tools/annotation-with-polygons/snap-tools/

Reviewer: The old `automatic-borders.md` page owned `/docs/manual/advanced/annotation-with-polygons/automatic-borders/` and `/docs/annotation/tools/annotation-with-polygons/automatic-borders/`. Since this PR deletes that page, those aliases should probably be preserved here as well, otherwise existing links/bookmarks will break.

Author: Makes sense.

---

# Snap to contour

Snap to contour allows you to automatically trace the outline of existing shapes when drawing or editing a polygon.
This feature helps you quickly create precise annotations by snapping to the edges of nearby objects.

You can enable snap to contour in several ways:

- **Using the Snap Tools control** : Click the Snap Tools button in the annotation sidebar,
  then toggle the "Snap to contour" button.

![Snap to contour button](/images/snap_to_contour_button.png)

> - **Keyboard shortcut**: Go to shortcut settings and set up a shortcut for a “Snap to contour" feature

Reviewer: This sentence still looks unfinished: the quotes are mismatched (`“Snap to contour"`), and it would be good to keep the wording exactly aligned with the UI shortcut text (`Toggle snap to contour`). A quick copy pass over this page would help.

Author: Fixed the quote. Renaming to “exact align” feels unnecessary, since we’re referring to the feature itself here, not the setting.

---
title: "Snap tools"
linkTitle: "Snap tools"
weight: 2
aliases:
- /docs/manual/advanced/annotation-with-polygons/snap-tools/
- /docs/annotation/tools/annotation-with-polygons/snap-tools/
---

# Snap to contour

> ![Example of annotation made with polygon and snap to contour option](/images/gif025_mapillary_vistas.gif)

Reviewer 2: this gif is outdated

Author: Replaced with new gif

Snap to contour allows you to automatically trace the outline of existing shapes when drawing or editing a polygon.
This feature helps you quickly create precise annotations by snapping to the edges of nearby objects.

You can enable snap to contour in several ways:

- **Using the Snap Tools control** : Click the Snap Tools button in the annotation sidebar,
  then toggle the "Snap to contour" button.

![Snap to contour button](/images/snap_to_contour_button.png)

> - **Keyboard shortcut**: Go to shortcut settings and set up a shortcut for a  “Snap to contour" feature

Reviewer 2: ```suggestion
- **Keyboard shortcut**: Go to shortcut settings and set up a shortcut for a “Snap to contour" feature
```

## Using snap to contour

Once enabled:

1. Start drawing or editing a polygon.
2. Points of other shapes will be highlighted, indicating that the polygon can attach to them.
3. Define the part of the contour you want to trace.

   ![Annotation with highlighted part for repetition](/images/image157_mapillary_vistas.jpg)

4. Click on the first point of the contour segment.

   ![Annotation with first contour point highlighted](/images/image158_mapillary_vistas.jpg)

> 5. Then click on any point along the path you want to trace. The selected point will be highlighted in purple.

Reviewer 2: Not relevant, the logic has been updated

>    ![Annotation with highlighted middle point](/images/image159_mapillary_vistas.jpg)

Reviewer 2: Not relevant image

6. Click on the last point and the contour will be traced automatically between your points.

   ![Annotation with last contour point highlighted](/images/image160_mapillary_vistas.jpg)

Below you can see results with opacity and black stroke applied:

![Example of annotation with applied opacity and black stroke](/images/image064_mapillary_vistas.jpg)

# Snap to point

> Snap to point allows you to automatically align polygon points with the corners and vertices of existing shapes.

Reviewer 2: polygon or polyline

> Snap to point allows you to automatically align polygon points with the corners and vertices of existing shapes.

Reviewer 2: ```suggestion
Snap to point allows you to automatically align polygon points with the vertices of existing shapes.
```

This feature helps you create precise annotations by snapping points to existing shape boundaries,
ensuring perfect alignment between adjacent objects.

## Enabling snap to point

You can enable snap to point in several ways:

- **Using the Snap Tools control**: Click the Snap Tools button in the annotation sidebar,
  then toggle the "Snap to point" button.

![Snap to point button](/images/snap_to_point_button.png)

> - **Keyboard shortcut**: Go to shortcut settings and set up a shortcut for the "Snap to point" feature.

Reviewer 2: <img width="506" height="275" alt="Image" src="https://github.com/user-attachments/assets/9b84109c-80e7-4171-9811-fbaad28be073" />

Let's use one word "switch" or "toggle"

Reviewer 3: "switch"  [means](https://www.merriam-webster.com/dictionary/switch#dictionary-entry-2) "to go from one thing to another in a set of equal things"
It's better to use "toggle" in both cases since it [means](https://www.merriam-webster.com/dictionary/toggle#dictionary-entry-2) "to turn a setting on or off", usually used in binary states

Author: Applied

## Using snap to point

Once enabled:

> 1. Start drawing or editing a polygon, polyline, or points shape.

Reviewer 2: Probably "editing may be confusing here, as is a special mode when we may redraw part of a polygon/polyline or points"

```suggestion
1. Start drawing or moving a point of a polygon, polyline, or points shape.
```

> 2. As you move your cursor near existing shape vertices, a snap indicator will appear.

Reviewer 2: What is `snap indicator`?

Author: Removed
